src/tests/electron/common/view-controllers/app-controller.ts

```diff
@@ -71,4 +68,10 @@ export class AppController {
             } class high-contrast-theme`,
         );
     }
+
+    public async setTelemetryState(enableTelemetry: boolean): Promise<void> {
+        await this.app.webContents.executeJavaScript(
+            `window.insightsUserConfiguration.setTelemetryState(${enableTelemetry})`,
```

Comment:
This might be happening too early, since it's going before we wait for a view to be visible; maybe better would be for it to involve some sort of `waitUntil` that verifies `window.insightsUserConfiguration` is defined first?

Reply:
that's a good point. I'll work on it.

Reply:
Updated. Is this close to what you had in mind?

Reply:
Yes, this seems about right to me!